packages/node/src/transport/WsInboundTransport.ts

```diff
@@ -1,22 +1,17 @@
-import type { Agent, InboundTransporter, Logger, OutboundPackage } from '../../packages/core/src'
-import type { TransportSession } from '../../packages/core/src/agent/TransportService'
+import type { Agent, InboundTransporter, Logger, OutboundPackage, TransportSession } from '@aries-framework/core'
 
-import WebSocket from 'ws'
+import { AriesFrameworkError, AgentConfig, TransportService, utils } from '@aries-framework/core'
+import WebSocket, { Server } from 'ws'
 
-import { AriesFrameworkError } from '../../packages/core/src'
-import { AgentConfig } from '../../packages/core/src/agent/AgentConfig'
-import { TransportService } from '../../packages/core/src/agent/TransportService'
-import { uuid } from '../../packages/core/src/utils/uuid'
-
-export class WsInboundTransporter implements InboundTransporter {
-  private socketServer: WebSocket.Server
+export class WsInboundTransport implements InboundTransporter {
+  private socketServer: Server
   private logger!: Logger
 
   // We're using a `socketId` just for the prevention of calling the connection handler twice.
   private socketIds: Record<string, unknown> = {}
 
-  public constructor(socketServer: WebSocket.Server) {
-    this.socketServer = socketServer
+  public constructor({ server, port }: { server: Server; port?: undefined } | { server?: undefined; port: number }) {
```

Comment:
Would it work with the following types?
```suggestion
  public constructor({ server, port }: { server: Server; port?: string } | { server?: Server; port: number }) {
```

Reply:
I want it to be an XOR statement. So either port or server should be provided but not both. If you create the server yourself, you set the port on the server. If you don't create the server yourself, you need to pass the port.

Could probably improve, but this was the easiest way I could think of with explicit typing that you shouldn't provide both